### PR TITLE
Change angler scaling and most of the references

### PR DESCRIPTION
**Pull request by happyori**
As I was playing with Walk of Life - Rebirth (Brilliant mod overall, absolutely loved it :D) I found that angler's profession, outside of the lingering bauble effect is a straight nerf to the base game profession at least if we are not considering other mods for this and even then its not great.

## Reasoning

Angler profession in vanilla is 50% increase to fish sell price unlocked as soon as you select that profession.
There are 51 normal fishes (3 of which are night market / magic bait only) and 10 legendary ( 5 of which are only unlocked after walnut room is opened and extended familly quest is unlocked )
With 0.5% per normal and 2% per legendary getting max size on all of them we will reach 45.5% | (51 * 0.005 + 10 * 0.02) * 100
This by itself means that this is a 4.5% nerf to base game but thats only *after* we reach absolute perfection on ALL of the fishes in the base game.

Let explore a bit more, due to size of a fish being a random multiplier even if you get a perfect catch on the legendary fish (which is incredibly hard) you still run a chance of not getting the max size of that fish. This means that you basically need the prestige version of the angler profession to get 20% (all of the legendries) of the 45.5% multiplier. Prestige usually is super late game thing since you need to unlock mastery first.
At the point you do get all 45.5% multiplier you will very likely already have a SIGNIFICANTLY better way of making money (aka farming)

> Further more I believe that Night market fish needs to be caught with magic bait (post walnut room) to get max size. Referencing this [post](https://www.reddit.com/r/StardewValley/comments/16w762i/biggest_catch_a_guide_to_maximising_fish_size/). I have not tested if its doable now with 1.6 but I don't think much changed in that aspect.

This means that if we perfectly catch all the catchable fish prior to walnut room (which would require the entire year since some fish are season based) we would get total of 24% price multiplier aka 1% less then the Fisher profession in vanilla. Fisher profession is super easy to get within the first season and in some cases even within half of a first season.

TLDR; Angler profession is just too nerfed in my opinion even with the bauble effect lingering. It creates this toxic grinding for max size fish (which is partially luck based) to just get something we would normally get within the first season (Fisher profession) :( and not even talking about the vanilla Angler values

## Suggestions

Here are some of my suggestions to make this a bit more fun and instead of being a straight nerf make it more of a journey to get more value out of it than base game. What I mean by that is to start lower than vanilla angler and as player makes more progress with their fishing get more value then vanilla angler.

1. Double the multiplier per fish to 1% (and 4% for legendary respectively)
2. Add a flat increase to multiplier (like 25% to match fisher profession).

With option 1 we would have a significantly lower % in the start but we would be able to reach all the way to 91% later in the game. Again that means catching all of the legendary fish at max size which is not easy. Without legendary fish we would get to around 51%
> Option 1 is what this PR currently contains but I can update it to option 2 pretty quickly.

With option 2 we would start at around 25% and increase all the way up to 70.5% (which i would argue is still relatively small increase considering the farming buffs this mod implements but that's all personal preference)

Also I would suggest (and if this PR passes the initial check, I would be more then willing to implement it myself) putting the values as a slider in the config to be adjustable in game. For option 1 it would be per fish multiplier (and maybe legendary multiplier, 4x etc) and option 2 would be the manual offset.

## Conclusion

I think that considering everything else this mod does nerfing active fishing like this doesn't make sense (farming is objectively still way more powerful way of making money)

If we consider modded fishes this multiplier would be all over the place but since the limit to the multiplier is set in the config it should be fine even if we buff the % per fish.

PS: This PR is just a quick thing I cobbled together as an example of how it might work.

### CODE REVIEW

**Comment by daleao** [comment]
You will notice since you PR'd the code that legendary fish are not required to be at max size. So I think it's strange that a lot of your argument is based on that false assumption. (Also, my understanding was, and I may be wrong, that legendary fish are always automatically caught at max size anyway?)

You compared Angler to farming professions, but you didn't compare it to other fishing professions. How would this buff affect the balance between Angler and Aquarist?

**Comment by happyori** [comment]
Sorry about the legendary fish part, I honestly completely blanked out on that if statement XD.
Also maybe as a UX thing always mark caught legendary fish as "MAX" in collection to indicate that its being counted since thats probably what confused me.

I think that you are right that in comparison within the fishing professions Angler was always a more powerful choice. I always thought that it was the difference between "active" and "passive" playstyle. If "passive" playstyle is better than "active" why even try to do anything with "active", its more effort and clearly less reward.

Basically what spurred me to start this conversation was the feeling I got when I got Angler after playing vanilla.
I am an active fishing player and love the fishing minigame but now since both Angler and Artisan were reworked, while playing I just felt like I am wasting time trying to do anything with actively fishing and in the end abandoned it, which was not a great feeling imo. 

In any case I am just one guy wanting fishing to be more engaging and I love the idea that you have to participate more with fishing to gain the multiplier but I personally don't find it fun having the end goal of this little mini game of getting max size being a nerfed version of vanilla Angler hence my recommendations on scaling to start lower and rise higher.

On the topic of Aquarist I always thought that Ponds were just a reskinned crabpots with a few select benefits. It might be more interesting if the Fish Ponds were more refined and had a whole separate progression system. Like coops that can be upgraded maybe there could be something for fish ponds and Aquarist in there. Like as an example certain upgrades that would cause more roe to be produced at the expense of other materials not producing at all. I will be honest I am not at the stage where I played with fish ponds and Aquarism / Aquarist yet so I can't tell if there needs

**Comment by happyori** [comment]
Also wanted to mention that despite everything I do think that the other branches are a lot more balanced where I don't feel like I *have* to play a certain way.

**Comment by happyori** [comment]
I think when I get the time this weekend I will update the PR to instead allow players to change these values in settings to be more in line with balancing that the rest of the mod does (like the other balance sliders for other professions)

**Comment by happyori** [comment]
@daleao I added the config value for the scaling. Tested this out locally and its working just fine

I have a few questions since I never actually published any mods, mainly what to do with
1. manifest.json 
2. localizations

1. Manifest seems to be what is used for releasing so I didn't want to change it but it might be better to create a separate branch where you can pull this PR from so as to not mess up with your release schedule
2. I tested this out locally and it just fallbacks to english so at least there is that.
